File: PaceCalculator.py

class PaceCalculator:
    def __init__(self):
        #Zonen Faktoren
        self.Zone1Faktor = 1.35
        self.Zone2Faktor = 1.22
        self.Zone3Faktor = 1.09
        self.Zone4Faktor = 1.02
        self.Zone5Faktor = 0.95

        #Zone Pace set auf 0 am Anfang
        self.Zone1Pace = 0
        self.Zone2Pace = 0
        self.Zone3Pace = 0
        self.Zone4Pace = 0
        self.Zone5Pace = 0

    def pace_zonen(self,minuten,sekunden):

        gesamt_sekunden = minuten * 60 + sekunden

        self.Zone1Pace = gesamt_sekunden * self.Zone1Faktor
        self.Zone2Pace = gesamt_sekunden * self.Zone2Faktor
        self.Zone3Pace = gesamt_sekunden * self.Zone3Faktor
        self.Zone4Pace = gesamt_sekunden * self.Zone4Faktor
        self.Zone5Pace = gesamt_sekunden * self.Zone5Faktor

        self.printPace()

    def printPace(self):
        print(f"Zone 1 (Erholung): {int((self.Zone1Pace)/60)}:{int((self.Zone1Pace%60))}/km")
        print(f"Zone 2 (Grundlagenausdauer):  {int((self.Zone2Pace)/60)}:{int((self.Zone2Pace%60))}/km")
        print(f"Zone 3 (Tempo):  {int((self.Zone3Pace)/60)}:{int((self.Zone3Pace%60))}/km")
        print(f"Zone 4 (Laktat-Akkumulation):  {int((self.Zone4Pace)/60)}:{int((self.Zone4Pace%60))}/km")
        print(f"Zone 5 (VO2max):  {int((self.Zone5Pace)/60)}:{int((self.Zone5Pace%60))}/km")

    def inputPace(self):
        print("---------------------------------")
        minuten = int(input("Minuten: "))
        sekunden = int(input("Sekunden: "))
        print("---------------------------------")
        self.pace_zonen(minuten,sekunden)



calculator = PaceCalculator()
calculator.inputPace()

File: PaceCalculator.java
import java.util.Scanner;
public class PaceCalculator {
    //Zeit
    private double gesamt_sekunden = 0;
    private double minuten = 0;
    private double sekunden = 0;

    //Zonen Faktoren
    private double Zone1Faktor = 1.35;
    private double Zone2Faktor = 1.22;
    private double Zone3Faktor = 1.09;
    private double Zone4Faktor = 1.02;
    private double Zone5Faktor = 0.95;

    //Zone Pace set auf 0 am Anfang
    private double Zone1Pace = 0;
    private double Zone2Pace = 0;
    private double Zone3Pace = 0;
    private double Zone4Pace = 0;
    private double Zone5Pace = 0;

    //Konstruktor
    public PaceCalculator(){

    }

    public void pace_zonen(){
        this.gesamt_sekunden = this.minuten * 60 + this.sekunden;

        //Berechung der Pace-Zonen in Sekunden pro Kilometer
        this.Zone1Pace = this.gesamt_sekunden * this.Zone1Faktor;
        this.Zone2Pace = this.gesamt_sekunden * this.Zone2Faktor;
        this.Zone3Pace = this.gesamt_sekunden * this.Zone3Faktor;
        this.Zone4Pace = this.gesamt_sekunden * this.Zone4Faktor;
        this.Zone5Pace = this.gesamt_sekunden * this.Zone5Faktor;

        this.printPace();
    }

    public void printPace() {
        System.out.println("Zone 1 (Erholung): " + (int) this.Zone1Pace/60 +":"+(int) this.Zone1Pace%60);
        System.out.println("Zone 2 (Grundlagenausdauer): " + (int) this.Zone2Pace/60 +":"+(int) this.Zone2Pace%60);
        System.out.println("Zone 3 (Tempo): "+ (int) this.Zone3Pace/60 +":"+(int) this.Zone3Pace%60);
        System.out.println("Zone 4 (Laktat-Akkumulation): "+ (int) this.Zone4Pace/60 +":"+(int) this.Zone4Pace%60);
        System.out.println("Zone 5 (V02Max): "+ (int) this.Zone5Pace/60 +":"+(int) this.Zone5Pace%60);
    }

    public void inputPace(){
        Scanner scan = new Scanner(System.in);
        System.out.println("----------------------------------------");
        System.out.print("Minuten: ");
        this.minuten = scan.nextInt();
        System.out.print("\nSekunden: ");
        this.sekunden = scan.nextInt();
        System.out.println("----------------------------------------");
        this.pace_zonen();
    }
}


public class Main {
    public static void main(String[] args) {
        PaceCalculator p1 = new PaceCalculator();
        p1.inputPace();
    }
}
